### PR TITLE
The API update

### LINKED ISSUE #40: [Feature] Autoclose Warning
Hello,

On this moment a ticket made in #tech-support gets closed after **one day** if no message has been send send since the ticket got openend, it will close.

If messages are being send, **two days** after the last message the ticket will be closed.

Is it possible to inform the user, before the automatic close, that it will be closed? This might be helpful for people have a thread open but forgot about it.

_My English is bad today, sorry!_

I did totally not copy and paste this message from Discord. ;-)

### LINKED ISSUE #47: [Enhancement]: Actual Tickets Integration
### Summary

The feature of future! We will actually get data from the main Tickets. Bot involves another whole batch of annoyances and so is planned for v0.1.1.

### Description

Basically we will have to make an API for the Tickets bot to send POST requests to.
We would then get this great package:
{
  "guild_id": "508392876359680000",
  "user_id": "585576154958921739",
  "ticket_id": 123,
  "ticket_channel_id": "508410703439462400"
}

### The Basis

This would remove the need for ticket users and change the mechanism based on which we run our opening actions. Making the code more reliable for the intents and purposes.

### Use Case

We can get user_id reliably. As it currently stands we have no practical way for doing so.

### Impact

A lot of work. Reliability. Less setup? They would have to add the integration on panel.ticketsbot.net

### Alternatives

The one is currently in use yes.

### Screenshots

_No response_

### Additional Information

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by Tech-TTGames**
## Description

Do get the user and actually only bot-channels. More efficient.

## Changes Made

- Integration endpoint implemented
- API created

## Issues Resolved

- Closes #47 
- Closes #40 

## Checklist

- [x] My code follows the style guidelines of this project.
- [x] I have reviewed my own code and ensured it passes all linting tests.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.

### CODE REVIEW

**Comment by Tech-TTGames** [comment]
Github had a stroke hmm?

File: tickets_plus/cogs/routines.py [hunk 1]
```diff
@@ -91,6 +92,51 @@ async def before_clean_status(self):
         """
         await self._bt.wait_until_ready()
 
+    @tasks.loop(minutes=2, seconds=30)
```
Comment (benfoster04) [inline tickets_plus/cogs/routines.py:95]: Potentially worth making these more configurable if the bot is self-hostable? Eg. Putting variables into config.

**Review by benfoster04 — CHANGES_REQUESTED** [review]

File: tickets_plus/api/handlers.py [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+# If later approved by the Initial Contrubotor, GPL-3.0-or-later.
+
+import json
+
+import discord
+from tornado import web
+from sqlalchemy import orm
+
+from tickets_plus import bot
+from tickets_plus.cogs import events
```
[the remark this answers is not in the document]
Comment (Tech-TTGames) [reply tickets_plus/api/handlers.py:30]: @benfoster04 already fixed that per se.

File: tickets_plus/cogs/routines.py (hunk 1 above)
Comment (Tech-TTGames) [reply tickets_plus/cogs/routines.py:95]: Hmm. Maybe something like runtimeconfig.json ? For more 'advanced' settings that have reasonable defaults. Wouldn't put this in the normal config.

File: example_secret.json [hunk 3]
```diff
@@ -1,4 +1,7 @@
 {
```
[the remark this answers is not in the document]
Comment (Tech-TTGames) [reply example_secret.json:1]: No, I'll just expand the setup to cover the new data needed.

File: example_config.json [hunk 4]
```diff
@@ -1,17 +1,23 @@
 {
```
[the remark this answers is not in the document]
Comment (Tech-TTGames) [reply example_config.json:1]: https://github.com/Tech-TTGames/Tickets-Plus/pull/77#discussion_r1181602204

File: tickets_plus/cogs/routines.py (hunk 1 above)
Comment (Tech-TTGames) [reply tickets_plus/cogs/routines.py]: Done.

**Review by benfoster04 — APPROVED** [review]
LGTM and cleared for merge. Good work! Looks consistent and a very good implementation to the bot structure.

**Pull request merged**